### ISSUE TITLE
[TextInputLayout] Add isActivated state for box stroke

### ISSUE THREAD

**Issue by malloth** [material-components/material-components-android#1776]
**Is your feature request related to a problem? Please describe.**
Currently internally TextInputLayout doesn't use the ColorStateLists like other views do (but only ints) and therefore we can not set some states (**isActivated**, **isChecked** and custom ones) like we are able to do in other views.

**Describe the solution you'd like**
Please use ColorStateLists instead of Ints internally and add a support for **isActivated** state.

**Describe alternatives you've considered**
None. This view looks like it was created by a junior developer, who didn't have any idea how all views are being made in Android OS. Please assign a senior and do it a right way (not joking).

**Additional context**
None.

**Comment by ornolisvt** [comment]
Yes, please add the possibility of adding new drawable states and setting the corresponding stroke color. TextInputLayout is not open for extension violating well known programming principles
